File: python/ascii-table.py
# Python program that prints ascii table values between two integers

try:
    start = int(input("Enter the starting number: "))
    limit = int(input("Enter the limit number: "))

    if limit < start:
        raise ValueError

except ValueError:
    print("Invalid value(s)")
    quit()

# Prints a table containing the number in decimal, hexadecimal, binary, then the char it represents
print("DEC\t\tHEX\t\tBIN\t\tCHAR")
for i in range(start, limit + 1):
    print(f"{i}\t\t", end="")
    print(f"{hex(i)}\t\t", end="")
    print(f"{bin(i)}\t\t", end="")
    print(chr(i))

File: ts/asciiTable.ts
function range(start: number, end: number) {
  // https://stackoverflow.com/a/19506234/14900882
  return Array.apply(0, Array(end - 1)).map(
    (_element: any, index: number) => index + start
  );
}

function asciiTable(start: number, end: number) {
  if (end < start) {
    throw new Error("Ending number cannot be smaller than starting number.");
  }

  console.log("DEC\t\tHEX\t\tBIN\t\tCHAR");

  for (let i in range(start, end)) {
    console.log(`${i}\t\t${String.fromCharCode(parseInt(i))}`);
  }
}

asciiTable(1, 3);
asciiTable(3, 1);
